digest/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+use multihash::Multihash;
+
+pub fn digest(bytes: &[u8]) -> Multihash {
+    // Time the tests and take fastest hash?
+    multihash::Sha3_256::digest(bytes)
+}
+
+pub trait DigestRoot {
```

Comment:
This trait needs renaming IMO.  It took me a while to workout exactly _why_.  As commented before `Digest` trait is good - its a noun.  But `DigestRoot` is using the word 'digest' as a verb, in crypto 'digest' is a noun not a verb - it is the output of a hash function.  Can we re-name this to be a noun?  I'm struggling to find something nice and short; 'FieldSortedDigest', 'OrderedDigest', 'FieldOrderedDigest'

Reply:
Oh, this could just become `RootDigest` and then 'digest' is a noun.